geometric_kernels/kernels/feature_maps.py

```diff
@@ -18,32 +20,22 @@
 )
 
 
-def deterministic_feature_map_compact(
-    space: DiscreteSpectrumSpace,
-    num_levels: int,
-):
-    r"""
-    Deterministic feature map for compact spaces based on the Laplacian eigendecomposition.
-
-    :param space: space.
-    :param num_levels: number of levels in the kernel approximation.
-
-    :return: Callable
-        Signature: (X, params,  **kwargs)
-        :param X: [N, D] points in the space to evaluate the map on.
-        :param params: parameters of the kernel (lengthscale and smoothness).
-        :param normalize: normalize to have unit average variance (if omitted
-                          or None, follows the standard behavior of
-                          MaternKarhunenLoeveKernel).
-        :param **kwargs: unused.
-
-        :return: `Tuple(None, features)` where `features` is [N, O] features.
-    """
+class DeterministicFeatureMapCompact:
```

Comment:
I suggest we create and abstract base class `FeatureMap` and inherit all the feature map classes from it. We used to use a `FeatureMap` alias for type hiniting before, this would serve as a replancement.